ArduCopter/GCS_Mavlink.cpp

```diff
@@ -836,6 +836,45 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
         mavlink_msg_command_int_decode(msg, &packet);
         switch(packet.command)
         {
+            case MAV_CMD_DO_SET_HOME: {
+                // assume failure
+                result = MAV_RESULT_FAILED;
+                if (is_equal(packet.param1, 1.0f)) {
+                    // if param1 is 1, use current location
+                    if (copter.set_home_to_current_location(true)) {
+                        result = MAV_RESULT_ACCEPTED;
+                    }
+                    break;
+                }
+                // sanity check location
+                if (!check_latlng(packet.x, packet.y)) {
```

Comment:
maybe move check_latlng after frame check, for when we later add non-global pos frames